ras_party/controllers/account_controller.py

```diff
@@ -389,18 +389,34 @@ def _send_account_email_changed_notification(email_address, new_email_address, r
     logger.info("Verification email sent for changing respondents email", respondent_id=str(respondent.party_uuid))
 
 
-@with_query_only_db_session
-def verify_token(token, session):
+def decode_token(token):
+    """
+    Decode the email verification token.
+    :param token: The email token to decode
+    :return: The decoded email address
+    """
+    duration = current_app.config["EMAIL_TOKEN_EXPIRY"]
     try:
-        duration = current_app.config["EMAIL_TOKEN_EXPIRY"]
         email_address = decode_email_token(token, duration)
     except SignatureExpired:
         logger.info("Expired email verification token")
         raise Conflict("Expired email verification token")
     except (BadSignature, BadData):
-        logger.exception("Bad token in verify_token")
+        logger.exception("Bad token in decode_token")
         raise NotFound("Unknown email verification token")
+    return email_address
+
+
+@with_query_only_db_session
+def verify_token(token, session):
+    """
+    Verify the email token and check if that respondent exists.
+    :param token: The email token for verification
+    :param session: Database session
+    :return: Verification response
+    """
 
+    email_address = decode_token(token)
```

Comment:
You're still inside the decorator and have just refactored the function, put_email_verification should just be getting an email, the token decode should be done outside session scope

Reply:
Have moved the decode outside of the function now. Finding it difficult to test this change but believe the recent changes have solved it.